www/js/plugin/storage.ts

```diff
@@ -0,0 +1,182 @@
+import { getAngularService } from "../angular-react-helper";
+import { addStatReading, statKeys } from "./clientStats";
+import { displayErrorMsg, logDebug } from "./logger";
+
+const mungeValue = (key, value) => {
+  let store_val = value;
+  if (typeof value != "object") {
+    // Should this be {"value": value} or {key: value}?
```

Comment:
nit: now that we are rewriting, remove comment or write down solution. this is no longer an open question

Reply:
I'll remove since it looks self-explanatory that we are doing `{key: value}`